gradio/deploy_space.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import annotations
+
+import os
+import random
+import re
+import sys
+
+import huggingface_hub
+
+import gradio as gr
+
+
+def add_configuration_to_readme(repo_directory, readme_file) -> dict:
+    configuration = {}
+
+    dir_name = os.path.basename(repo_directory)
+    title = input(f"Enter Spaces app title [{dir_name}]: ") or dir_name
+    formatted_title = format_title(title)
+    if formatted_title != title:
+        print(f"Formatted to {formatted_title}. ")
+    configuration["title"] = formatted_title
+
+    app_file = None
+    for file in os.listdir(repo_directory):
+        file_path = os.path.join(repo_directory, file)
+        if not os.path.isfile(file_path) or not file.endswith(".py"):
+            continue
+
+        with open(file_path, encoding="utf-8", errors="ignore") as f:
+            content = f.read()
+            if "import gradio" in content:
+                app_file = file
+                break
+
+    app_file = (
+        input(f"Enter Gradio app file {f'[{app_file}]' if app_file else ''}: ")
+        or app_file
+    )
+    if not app_file or not os.path.exists(app_file):
+        raise FileNotFoundError("Failed to find Gradio app file.")
+    configuration["app_file"] = app_file
+
+    emoji_set = "🤯🤖🧠🐶👑💥🐮🎁🐙🦋"
+    default_emoji = random.choice(emoji_set)
+    configuration["emoji"] = (
+        input(f"Enter Spaces Card emoji [{default_emoji}]: ") or default_emoji
+    )
+
+    color_set = ["red", "yellow", "green", "blue", "indigo", "purple", "pink", "gray"]
+    default_color = random.choice(color_set)
+    color = input(f"Enter Spaces Card color [{default_color}]: ") or default_color
+    configuration["colorFrom"] = color
+    configuration["colorTo"] = color
+    configuration["sdk"] = "gradio"
+    configuration[
+        "python_version"
+    ] = f"{sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}"
+    configuration["sdk_version"] = gr.__version__
+    huggingface_hub.metadata_save(readme_file, configuration)
```

Comment:
Throws an error on Windows: `UnicodeEncodeError: 'charmap' codec can't encode character '\U0001f916' in position 46: character maps to <undefined>`

Thi

Reply:
The huggingface_hub.metadata_save line? Can you file an issue on https://github.com/huggingface/huggingface_hub/

Reply:
PR here: https://github.com/huggingface/huggingface_hub/pull/1484/files

cc @Wauplin

Reply:
Thanks :pray: Merging it :)